app/src/main/res/values/strings.xml

```diff
@@ -232,13 +232,13 @@
     <string name="unsupport_magisk_msg">This version of the app does not support Magisk version lower than %1$s.\n\nThe app will behave as if no Magisk is installed, please upgrade Magisk as soon as possible.</string>
     <string name="unsupport_general_title">Abnormal State</string>
     <string name="unsupport_system_app_msg">Running this app as a system app is not supported. Please revert the app to a user app.</string>
-    <string name="unsupport_other_su_msg">An \"su\" command that does not belong to Magisk is detected. Please remove the other unsupported su.</string>
```

Comment:
This should have been "a \"su\" command".
How something belongs to Magisk is not explained.
"unsupported su" does not explain what it refers to.

Reply:
I agree with "an" being incorrect, otherwise this makes more sense as currently written. Magisk is root so "su" can belong to it.

Reply:
@osm0sis And it should be "was detected" too. Doesn't fix much though. Instead of trying to decipher all undeniable errors, try to envision a user making use of the information conveyed.

Right now it is in the form of a riddle following a garden-path sentence.
Only at the end of it all is there anything actionable.
What is "a su"? A command, the app requesting it, the app handling it?
Why is it unsupported?
What constitutes belonging to Magisk.

Reply:
You knowing what root is should be somewhat of a prerequisite.

Reply:
Ideally, yes. However, let's try to not throw concepts around and try to explain better where possible what is going on.
The outcome here is potentially disastrous.

This should be clear after reading the sentence, and we can increase chances of it happening:
* What to do
* what happened
* what it entailed
* what the resulting action/outcome should be

Regular people donate money too.

Reply:
Your proposed solution is still sub-optimal. And I don't think this one error message will have any effect on donations so that's neither here nor there.

Reply:
Why is it sub-optimal, and what meaning is lost?

This is the type of string that translators will either skip, or the point at which they will stop translating alltogether.
That leads to different parts translated by different people, incomplete translations.
The lacking quality and vagueness leads to confusion over what is what, long before it enters translation.

Nobody wants to donate to something that conveys a message of not being for them.
The ~"you should have just known this" is exclusive to a limited set of users, when there is no need for it.
I among those have a harder time grasping it.

The basic error of a/an and confusion over time doesn't seem professional.
Consider that this happens when something potentially disastrous is going on.

Failing graciously and instilling confidence drives donations, the original is neither of those.

Reply:
It's a technical error message involving 2 competing root solutions being installed simultaneously. The original is clunky but yours is not an improvement.

So, to convey the correct meaning, instead of:
```
An \"su\" command that does not belong to Magisk is detected. Please remove the other unsupported su.
```

Something like:
```
A \"su\" command not from Magisk has been detected. Please remove any competing root solution.
```

Reply:
Good, but the request being made is lost. If that isn't important, how about:

>"Please uninstall the competing rooting solution installed on your device."?

Then the user doesn't have to know what \"su\" is, or a binary is.

Reply:
Then the why is lost, the su binary part is important.

Reply:
It is an improvement for sure, but you lost me at "because the why is important"
I am thinking
"Please remove the competing rooting solution on your device by uninstalling it."
says the same thing.

Reply:
It being because of "su" is not covered. I believe this is meant to cover pre-installed su in ROMs and if someone tried to run Magisk app over SuperSU su. Knowing su is what triggered the error is extremely helpful for the first case, since it might not be as obvious. Then the second sentence covers both.

Reply:
Now I understand.
I think "uninstalling" alludes to it being an app moreso than "binary", seeing as that could be just a file.
If the user removes just the system file, a lot of stuff stops working and there is more support.
"On your device" makes it clear that it isn't external access of some sort.

Reply:
Right, so my suggestion is still more complete:

```
A \"su\" binary not from Magisk has been detected. Please remove any competing root solution.
```

Reply:
What makes it more complete (other than the new latter part)?

Reply:
It mentions the su binary was detected without being overly wordy, which I've continually told you is important. Then it tells them to remove root and/or reinstall Magisk. It's all succinct and accurate.

Reply:
It can't possibly be succinct when there is a disconnect between the two sentences _and_ it relies on inferring the relationship.

Notwithstanding "and/or reinstall Magisk"

>_A \"su\" binary not from Magisk has been detected. Please remove any competing root solution._

has more words and is longer than 

>"Please uninstall the competing \"su\" rooting solution installed on your device."

Reply:
I don't think I have the energy to relitigate this with you for the Nth time. I've explained how and why it needs to be, fix it as I said so we can move on and eventually merge this.

Reply:
Sorry, I changed the suggestion a while back thinking you would see it.
Now I changed the translation. No disconnect, nothing inferred, and what reinstalling Magisk does is explained.